api/custom_auth/oauth/serializers.py

```diff
@@ -85,12 +83,10 @@ def _get_user(self, user_data: dict):
 
         if not existing_user:
             sign_up_type = self.validated_data.get("sign_up_type")
-            if not (
-                settings.ALLOW_REGISTRATION_WITHOUT_INVITE
-                or sign_up_type == SignUpType.INVITE_LINK.value
-                or Invite.objects.filter(email=email).exists()
-            ):
-                raise PermissionDenied(USER_REGISTRATION_WITHOUT_INVITE_ERROR_MESSAGE)
+            if not settings.ALLOW_REGISTRATION_WITHOUT_INVITE:
```

Comment:
Do you think it makes sense to move the setting check inside the `_validate_registration_invite` method?

Reply:
Yeah, I debated this myself. I've moved it into the method as suggested.